src/suffix.c

```diff
@@ -403,13 +396,9 @@ void deleteSuffixTrieMap(TrieMap *trie, const char *str, uint32_t len) {
   // iterate all matching terms and remove word
   for (int j = 0; j < len - MIN_SUFFIX + 1; ++j) {
     suffixData *data = TrieMap_Find(trie, str + j, len - j);
-    // suffix trie is shared between all tag fields in index, even if they don't use it.
-    // if the trie is owned by other fields and not any one containing this suffix,
-    // then failure to find the suffix is not an error. just move along.
-    if (data == TRIEMAP_NOTFOUND) continue;
-    // RS_LOG_ASSERT(data != TRIEMAP_NOTFOUND, "all suffixes must exist");
+    RS_LOG_ASSERT(data != TRIEMAP_NOTFOUND, "all suffixes must exist");
     if (j == 0) {
-      // keep pointer to word string to free after it was found in al sub tokens.
+      // keep pointer to word string to free after it was found in all sub tokens.
```

Comment:
This reverts the changes done in #3289 which are relevant for the TEXT index, but not for the TAG.

Reply:
I suggest to add a test such as `testSuffixCleanup` from #3289 with a `TAG` field with `WITHSUFFIXTRIE`

Reply:
Why not have this change and the related test in a separate PR?